### ISSUE TITLE
Ship Stuck in Planet

### ISSUE THREAD

**Issue by SimonC4** [MovingBlocks/DestinationSol#59]
I was in the imperialBig ship on the first planet, the rocky planet, and got stuck inside the planet. Kind of sucked in and then the ship disappeared and the camera kept drifting to the middle of the planet. 

Seems like it's happened a few times before: http://steamcommunity.com/app/342980/discussions/1/606068060828393540/

**Comment by Cervator** [comment]
Yup have heard about this before, it just seems really tricky to reproduce and thus troubleshoot further.

Had coincidentally mentioned it to @grauerkoala earlier today in relation to #58 since it is probably some obscure bug like that which can cause it (but not that one)

**Comment by Cervator** [comment]
Another report - suggests (again) maybe the Vanguard (aka imperialBig) in particular can hit this issue. Problem with its hit box?

http://steamcommunity.com/app/342980/discussions/1/520518053428477307/

**Comment by Cervator** [comment]
More details from the last post, with possible reproduction instructions: 

"go to a flat spot on a planet then move your mouse in a clockwise circle around the ship for a minute or two thats how it worked for me"

So maybe you can sort of "wiggle" your way through the surface with a Vanguard.

**Comment by Viech** [comment]
This just happened to me with the Beetle on `1.3.0`, so it's not Vanguard-specific.

**Comment by Cervator** [comment]
Thanks for the extra detail @Viech :-)

**Comment by Cervator** [comment]
And another: http://steamcommunity.com/app/342980/discussions/1/520519112496560688/

Tendency for ships to "jump" slightly / be able to skip physics/collissions briefly? Seems related.

**Comment by crazywolf132** [comment]
What if you changed the rendering of the planet, so the it is no-longer renders as lots of little objects... what if you tried to do something more along the lines of pre-made planets. This would work by joining together the images and make a single .png and then load and scale that into the game.

**Comment by crazywolf132** [comment]
Anyone got an image of this?

**Comment by OxidizeMonk** [comment]
I got this problem too - when the tower on the snow planet shot me with a paralysing weapon, i teleported in the ground, under the tower.

**Comment by crazywolf132** [comment]
Ok thank you @OxidizeMonk.
Out of respect to the developers, here is how to fix it.   @Cervator.
You need to re-do the collision boxes, probably make them 3 pixels bigger than what they need to be.
This will compensate for the code not joining them to pixel perfection.

**Comment by Cervator** [comment]
Thanks @crazywolf132. Is that something you've tested in your fork, or just a theory? Did you redo your collision shapes successfully using some fixed version of the Box2D editor? Is it accessible somewhere?

**Comment by crazywolf132** [comment]
Our fixed box2D we lost... We got crypto Virus. As far as i am aware it is what we tested in our fork, but seen we are re-basing it, it wont be there. I will do some work on it tonight and get back to you with the result and the code if it checks out and doesn't cause any bugs.

**Comment by Cervator** [comment]
Another screenie of it happening via Steam:

http://steamcommunity.com/sharedfiles/filedetails/?id=884620460

From http://steamcommunity.com/app/342980/discussions/1/133259956004535197/

**Comment by crazywolf132** [comment]
Do you have a log for that @Cervator ?

**Comment by Cervator** [comment]
Old issue, still happens. Got a screenie for once and thought it interesting you can see a tiny spot of sky in the ground - seems highly likely that the "ground chunks" are being placed with gaps that are too large at times letting there be incomplete collision shapes that ships can sneak through (as usual this was an imperial large)

<img width="684" alt="2018-12-23 20_31_42-destination sol" src="https://user-images.githubusercontent.com/1063833/50390004-5096d480-06ff-11e9-9913-cc25c80c299f.png">
